examples/threathunter/threat_intelligence/threatintel.py

```diff
@@ -0,0 +1,240 @@
+import logging
+from feed_helper import FeedHelper
+from cabby import create_client
+from cbapi.psc.threathunter import CbThreatHunterAPI, Report
+from cbapi.errors import ApiError
+from cbapi.psc.threathunter.models import Feed
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+class ThreatIntel:
+    def __init__(self):
+        self.cb = CbThreatHunterAPI()
+
+    def push_to_cb(self, feed_id, results):
+        try:
+            feed = self.cb.select(Feed, feed_id)
+        except ApiError as e:
+            log.error(f"couldn't find CbTH feed {feed_id}: {e}")
+            return
+
+
+        report_list_to_send = []
+        report_dict_list_to_validate = []
+
+        #optional_report_attributes = ["link", "tags", "iocs", "iocs_v2", "visibility"]
+
+        for result in results:
+            try:
+                report_dict = { #document these values and what a result should contain
+                    "id": str(result.id),
+                    "timestamp": int(result.timestamp.timestamp()),
+                    "title": str(result.title),
+                    "description": str(result.description),
+                    "severity": int(result.severity),
+                    "iocs_v2": [ioc_v2.as_dict() for ioc_v2 in result.iocs_v2]
+                }
+
+
+                # for opt in optional_report_attributes:
+                #     if hasattr(result, opt):
+                #         if 'opt' == 'iocs_v2' or 'opt' == 'iocs':
+                #             report_dict[opt] = [ioc_v2.as_dict() for ioc_v2 in result.iocs_v2]
+                #         else:
+                #             report_dict[opt] = getattr(result, opt)
+
+                report_dict_list_to_validate.append(report_dict)
+
+                report = Report(self.cb, initial_data=report_dict, feed_id=feed_id)
```

Comment:
Shouldn't we validate the report prior to passing it to CBAPI as this may cause an error if something is not valid?

Reply:
Reports are validated in bulk, and aren't sent if validation fails. Line 49 here is just the creation of the Report object, but doesn't include sending it to CB. That happens below where there's a check for validation.